core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.kt

```diff
@@ -0,0 +1,284 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.bookmark
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.CompoundButton
+import android.widget.ImageView
+import android.widget.Switch
+import android.widget.TextView
+import androidx.appcompat.view.ActionMode
+import androidx.appcompat.widget.SearchView
+import androidx.appcompat.widget.Toolbar
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.AdapterDataObserver
+import butterknife.BindView
+import com.google.android.material.snackbar.Snackbar
+import org.kiwix.kiwixmobile.core.Intents.internal
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R2
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.extensions.setBitmapFromString
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.DialogShower
+import org.kiwix.kiwixmobile.core.utils.EXTRA_CHOSE_X_FILE
+import org.kiwix.kiwixmobile.core.utils.EXTRA_CHOSE_X_TITLE
+import org.kiwix.kiwixmobile.core.utils.EXTRA_CHOSE_X_URL
+import org.kiwix.kiwixmobile.core.utils.KiwixDialog
+import java.util.ArrayList
+import javax.inject.Inject
+
+class BookmarksActivity : BaseActivity(),
+  BookmarksContract.View, BookmarksAdapter.OnItemClickListener {
+  private val bookmarksList: MutableList<BookmarkItem> = ArrayList()
+  private val allBookmarks: MutableList<BookmarkItem> = ArrayList()
+  private val deleteList: MutableList<BookmarkItem> = ArrayList()
+
+  @JvmField @BindView(R2.id.toolbar)
+  var toolbar: Toolbar? = null
+
+  @JvmField @BindView(R2.id.recycler_view)
+  var recyclerView: RecyclerView? = null
+
+  @JvmField @BindView(R2.id.no_bookmarks)
+  var noBookmarks: TextView? = null
+
+  @JvmField @BindView(R2.id.bookmarks_switch)
+  var bookmarksSwitch: Switch? = null
+
+  @Inject
+  var presenter: BookmarksContract.Presenter? = null
+
+  @Inject
+  var zimReaderContainer: ZimReaderContainer? = null
+
+  @Inject
+  var dialogShower: DialogShower? = null
+  private var refreshAdapter = true
+  private var bookmarksAdapter: BookmarksAdapter? = null
+  private var actionMode: ActionMode? = null
+  private val actionModeCallback: ActionMode.Callback =
+    object : ActionMode.Callback {
+      override fun onCreateActionMode(
+        mode: ActionMode,
+        menu: Menu
+      ): Boolean {
+        mode.menuInflater.inflate(R.menu.menu_context_delete, menu)
+        bookmarksSwitch!!.isEnabled = false
+        return true
+      }
+
+      override fun onPrepareActionMode(
```

Comment:
check all these functions and if they can be on one line then make them be on one line

Reply:
@sonusourav please fix the functions that can be one one line